src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultSecretManager.cs

```diff
@@ -42,67 +43,71 @@ internal KeyVaultSecretManager(string keyVaultUrl, string azureClientId, string
             this.azureClientSecret = azureClientSecret;
         }
 
-        internal async Task<byte[]> GetHashedSecret(string secretName)
+        public async Task<byte[]> GetHashedSecret(string secretName, string secretVersion)
         {
-            if (secretCache.ContainsKey(secretName))
+            var combinedNameAndVersion = FormatSecretNameAndVersion(secretName, secretVersion);
+            if (secretCache.ContainsKey(combinedNameAndVersion))
             {
-                return secretCache[secretName];
+                return secretCache[combinedNameAndVersion];
             }
 
-            var secret = await GetSecretFromKeyVault(secretName);
+            var secret = await GetSecretFromKeyVault(secretName, secretVersion);
             using (var sha256 = SHA256.Create())
             {
                 var secretAsBytes = Encoding.UTF8.GetBytes(secret);
                 var hashedSecret = sha256.ComputeHash(secretAsBytes);
-                secretCache.Add(secretName, hashedSecret);
+                secretCache.Add(combinedNameAndVersion, hashedSecret);
```

Comment:
Considering that the code can be executed concurrently, should `secretCache` be a `ConcurrentDictionary`?